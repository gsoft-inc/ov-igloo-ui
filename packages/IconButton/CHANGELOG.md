# @igloo-ui/icon-button

## 1.0.4

### Patch Changes

- bf4ddef: Updated storybook version and linting. Made linting fixes in these components.
- Updated dependencies [bf4ddef]
  - @igloo-ui/button@0.5.2

## 1.0.3

### Patch Changes

- c3efaf5: Fixed tests and snapshots since react-aria generated ids changed and viewBox was added to icons.
- Updated dependencies [c3efaf5]
  - @igloo-ui/button@0.5.1

## 1.0.2

### Patch Changes

- Updated dependencies [357133d]
  - @igloo-ui/button@0.5.0

## 1.0.1

### Patch Changes

- a41e1d4: Fix: moved destructuring of props to function parameters so that storybook will automatically show the default value in the props table. It was done in Modal because it needed to send props to useDialog. In this case, @default annotation was used in comments.
- Updated dependencies [a41e1d4]
  - @igloo-ui/button@0.4.1

## 1.0.0

### Major Changes

- e47ae88: A secondary ghost appearance was added to the icon button.

  ## BREAKING CHANGE

  Since the ghost icon button was already grey, we needed to set the default back to blue and add the secondary variant, which is grey.

### Patch Changes

- Updated dependencies [d2e08a0]
- Updated dependencies [d5c91f5]
  - @igloo-ui/button@0.4.0

## 0.2.13

### Patch Changes

- 6ee9c45: Updated anything causing errors or warnings in the browser console
- Updated dependencies [6ee9c45]
  - @igloo-ui/button@0.3.7

## 0.2.12

### Patch Changes

- 0640409: Update @igloo-ui/icons to 1.6.0
- 0640409: udpate classnames to 2.3.2
- Updated dependencies [0640409]
- Updated dependencies [0640409]
  - @igloo-ui/button@0.3.6

## 0.2.11

### Patch Changes

- d971143: All tests were updated to use testing library and not enzyme
- Updated dependencies [d971143]
  - @igloo-ui/button@0.3.5

## 0.2.10

### Patch Changes

- 9842a0d: add forwardRef

## 0.2.9

### Patch Changes

- 38a4994: Update dependency on @igloo-ui/icons and other @igloo libs
- Updated dependencies [4de58ec]
- Updated dependencies [38a4994]
  - @igloo-ui/button@0.3.0

## 0.2.8

### Patch Changes

- 0a86f13: Update content of README
- Updated dependencies [0a86f13]
  - @igloo-ui/button@0.2.2

## 0.2.7

### Patch Changes

- Updated dependencies [f019f88]
  - @igloo-ui/button@0.2.0

## 0.2.6

### Patch Changes

- 8c8686e: Change @import of font.css file for a @use in SASS files
- Updated dependencies [8c8686e]
  - @igloo-ui/button@0.1.11

## 0.2.5

### Patch Changes

- e8f201c: Rename the output file for css

## 0.2.4

### Patch Changes

- eceb176: Fix the size of icon-button when a button is on the project

## 0.2.3

### Patch Changes

- 4a256e3: IE11 compatibility
- Updated dependencies [4a256e3]
  - @igloo-ui/button@0.1.9

## 0.2.2

### Patch Changes

- 1e1c6e5: Chore Update packages
- Updated dependencies [1e1c6e5]
  - @igloo-ui/button@0.1.8

## 0.2.1

### Patch Changes

- 0d8e80e: Remove useless background property already specified in button

## 0.2.0

### Minor Changes

- bd0ece1: Change IconButtonProps extends to use ButtonProps

## 0.1.2

### Patch Changes

- ec6cef7: Component IconButton: Fix styles naming in package and adjust README

## 0.1.1

### Patch Changes

- 2da8010: Fix render of elements with the css var --grey-{...}
- Updated dependencies [2da8010]
  - @igloo-ui/button@0.1.4

## 0.1.0

### Minor Changes

- 3af2f78: ## IconButton component

  Add IconButton component to create buttons with an SVG icon inside it.

  ### Component API

  | name      | description                                                                                     |
  | --------- | ----------------------------------------------------------------------------------------------- |
  | className | Add class names to the surrounding DOM container.                                               |
  | icon      | Icon React node to represent the value of the button                                            |
  | onClick   | Callback function that will be called when the user clicks on the button.                       |
  | disabled  | True if the control is disabled and shows a disabled state. The user cannot click on the button |
  | size      | Button size                                                                                     |
  | rounded   | True for rounded corners                                                                        |
  | dataTest  | Add a data-test tag for automated tests                                                         |
