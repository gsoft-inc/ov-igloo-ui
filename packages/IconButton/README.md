# @igloo-ui/icon-button

Welcome to my `icon-button` component.

## Installation

```sh
npm i @igloo-ui/icon-button
# or with yarn
yarn add @igloo-ui/icon-button
```

Then to use the component in your code just import it!

```js
import icon-button from '@igloo-ui/icon-button';
// and with css-modules
import '@igloo-ui/icon-button/dist/icon-button.css';
```
