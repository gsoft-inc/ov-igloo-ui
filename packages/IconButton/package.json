{
  "name": "@igloo-ui/icon-button",
  "version": "0.2.9",
  "main": "dist/IconButton.js",
  "module": "dist/IconButton.js",
  "types": "dist/IconButton.d.ts",
  "styles": "dist/icon-button.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "sylvester.vuong@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "dependencies": {
    "@igloo-ui/button": "^0.3.0",
    "@igloo-ui/tokens": "^1.1.1",
    "classnames": "^2.3.1"
  },
  "devDependencies": {
    "@igloo-ui/icons": "^1.3.0"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ]
}
