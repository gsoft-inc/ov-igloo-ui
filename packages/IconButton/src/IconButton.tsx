import * as React from 'react';
import cx from 'classnames';

import Button, { ButtonOwnProps } from '@igloo-ui/button';

import './icon-button.scss';

export type Size = 'xsmall' | 'small' | 'medium' | 'large';
export type Ref = HTMLButtonElement;

export interface IconButtonProps extends Omit<ButtonOwnProps, 'size'> {
  /** Add class names to the surrounding DOM container */
  className?: string;
  /** Icon React node to represent the value of the button */
  icon: React.ReactNode;
  /** Callback function that will be called
   * when the user clicks on the button */
  onClick?: () => void;
  /** True if the control is disabled and shows a disabled state.
   * The user cannot click on the button */
  disabled?: boolean;
  /** Button size */
  size?: Size;
  /** True for rounded corners */
  rounded?: boolean;
  /** Add a data-test tag for automated tests */
  dataTest?: string;
}

const IconButton: React.FunctionComponent<IconButtonProps> = React.forwardRef<
  Ref,
  IconButtonProps
>(
  (
    {
      className,
      icon,
      onClick,
      disabled = false,
      size = 'medium',
      rounded = false,
      dataTest,
      ...rest
    }: IconButtonProps,
    ref,
  ) => {
    const classes = cx('ids-icon-btn', `ids-icon-btn--${size}`, className, {
      'ids-icon-btn--round': rounded,
    });

    return (
      <Button
        ref={ref}
        className={classes}
        onClick={onClick}
        disabled={disabled}
        data-test={dataTest}
        {...rest}
      >
        {icon}
      </Button>
    );
  },
);

export default IconButton;
