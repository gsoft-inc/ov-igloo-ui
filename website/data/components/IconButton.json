[
  [
    {
      "tags": {},
      "filePath": "/Users/franck.gaudin/Devel/Igloo/components/packages/IconButton/src/IconButton.tsx",
      "description": "",
      "displayName": "IconButton",
      "methods": [],
      "props": {
        "className": {
          "defaultValue": null,
          "description": "Add class names to the surrounding DOM container.",
          "name": "className",
          "parent": {
            "fileName": "components/packages/IconButton/src/IconButton.tsx",
            "name": "IconButtonProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/IconButton/src/IconButton.tsx",
              "name": "IconButtonProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "icon": {
          "defaultValue": null,
          "description": "Icon React node to represent the value of the button",
          "name": "icon",
          "parent": {
            "fileName": "components/packages/IconButton/src/IconButton.tsx",
            "name": "IconButtonProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/IconButton/src/IconButton.tsx",
              "name": "IconButtonProps"
            }
          ],
          "required": true,
          "type": { "name": "ReactNode" }
        },
        "onClick": {
          "defaultValue": null,
          "description": "Callback function that will be called when the user clicks on the button.\n@returns",
          "name": "onClick",
          "parent": {
            "fileName": "components/packages/IconButton/src/IconButton.tsx",
            "name": "IconButtonProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/IconButton/src/IconButton.tsx",
              "name": "IconButtonProps"
            }
          ],
          "required": false,
          "type": { "name": "() => void" }
        },
        "disabled": {
          "defaultValue": null,
          "description": "True if the control is disabled and shows a disabled state. The user cannot click on the button",
          "name": "disabled",
          "parent": {
            "fileName": "components/packages/IconButton/src/IconButton.tsx",
            "name": "IconButtonProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/IconButton/src/IconButton.tsx",
              "name": "IconButtonProps"
            }
          ],
          "required": false,
          "type": { "name": "boolean" }
        },
        "size": {
          "defaultValue": null,
          "description": "Button size",
          "name": "size",
          "parent": {
            "fileName": "components/packages/IconButton/src/IconButton.tsx",
            "name": "IconButtonProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/IconButton/src/IconButton.tsx",
              "name": "IconButtonProps"
            }
          ],
          "required": false,
          "type": { "name": "Size" }
        },
        "rounded": {
          "defaultValue": null,
          "description": "True for rounded corners",
          "name": "rounded",
          "parent": {
            "fileName": "components/packages/IconButton/src/IconButton.tsx",
            "name": "IconButtonProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/IconButton/src/IconButton.tsx",
              "name": "IconButtonProps"
            }
          ],
          "required": false,
          "type": { "name": "boolean" }
        },
        "dataTest": {
          "defaultValue": null,
          "description": "Add a data-test tag for automated tests",
          "name": "dataTest",
          "parent": {
            "fileName": "components/packages/IconButton/src/IconButton.tsx",
            "name": "IconButtonProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/IconButton/src/IconButton.tsx",
              "name": "IconButtonProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "children": {
          "defaultValue": null,
          "description": "The content to display inside the button",
          "name": "children",
          "parent": {
            "fileName": "components/packages/Button/dist/Button.d.ts",
            "name": "ButtonOwnProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Button/dist/Button.d.ts",
              "name": "ButtonOwnProps"
            },
            {
              "fileName": "components/node_modules/@types/react/index.d.ts",
              "name": "TypeLiteral"
            }
          ],
          "required": false,
          "type": { "name": "ReactNode" }
        },
        "active": {
          "defaultValue": null,
          "description": "Sets the button in an active state",
          "name": "active",
          "parent": {
            "fileName": "components/packages/Button/dist/Button.d.ts",
            "name": "ButtonOwnProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Button/dist/Button.d.ts",
              "name": "ButtonOwnProps"
            }
          ],
          "required": false,
          "type": { "name": "boolean" }
        },
        "loading": {
          "defaultValue": null,
          "description": "Replaces button text with a spinner while a background action is being performed",
          "name": "loading",
          "parent": {
            "fileName": "components/packages/Button/dist/Button.d.ts",
            "name": "ButtonOwnProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Button/dist/Button.d.ts",
              "name": "ButtonOwnProps"
            }
          ],
          "required": false,
          "type": { "name": "boolean" }
        },
        "appearance": {
          "defaultValue": null,
          "description": "Button appearance",
          "name": "appearance",
          "parent": {
            "fileName": "components/packages/Button/dist/Button.d.ts",
            "name": "ButtonOwnProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Button/dist/Button.d.ts",
              "name": "ButtonOwnProps"
            }
          ],
          "required": false,
          "type": { "name": "Appearance" }
        },
        "iconLeading": {
          "defaultValue": null,
          "description": "Icon to display to the left of button content",
          "name": "iconLeading",
          "parent": {
            "fileName": "components/packages/Button/dist/Button.d.ts",
            "name": "ButtonOwnProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Button/dist/Button.d.ts",
              "name": "ButtonOwnProps"
            }
          ],
          "required": false,
          "type": { "name": "ReactNode" }
        },
        "iconTrailing": {
          "defaultValue": null,
          "description": "Icon to display to the right of button content",
          "name": "iconTrailing",
          "parent": {
            "fileName": "components/packages/Button/dist/Button.d.ts",
            "name": "ButtonOwnProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Button/dist/Button.d.ts",
              "name": "ButtonOwnProps"
            }
          ],
          "required": false,
          "type": { "name": "ReactNode" }
        },
        "showOnlyIconOnMobile": {
          "defaultValue": null,
          "description": "Display only the icon in mobile",
          "name": "showOnlyIconOnMobile",
          "parent": {
            "fileName": "components/packages/Button/dist/Button.d.ts",
            "name": "ButtonOwnProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Button/dist/Button.d.ts",
              "name": "ButtonOwnProps"
            }
          ],
          "required": false,
          "type": { "name": "boolean" }
        },
        "type": {
          "defaultValue": null,
          "description": "Optional prop to specify the type of the Button",
          "name": "type",
          "parent": {
            "fileName": "components/packages/Button/dist/Button.d.ts",
            "name": "ButtonOwnProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Button/dist/Button.d.ts",
              "name": "ButtonOwnProps"
            }
          ],
          "required": false,
          "type": { "name": "\"button\" | \"reset\" | \"submit\"" }
        },
        "intercomTarget": {
          "defaultValue": null,
          "description": "Add a data-intercom-target with unique id to link a component to a Product Tour step",
          "name": "intercomTarget",
          "parent": {
            "fileName": "components/packages/Button/dist/Button.d.ts",
            "name": "ButtonOwnProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Button/dist/Button.d.ts",
              "name": "ButtonOwnProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        }
      }
    }
  ]
]
