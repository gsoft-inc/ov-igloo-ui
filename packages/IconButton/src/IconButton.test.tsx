/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';
import Plus from '@igloo-ui/icons/dist/Plus';

import IconButton from './IconButton';

const setUp = (props = {}) => {
  const component = shallow(
    <IconButton
      icon={<Plus size="small" />}
      dataTest="ids-icon-btn"
      {...props}
    />
  );
  return component;
};

describe('IconButton Component', (): void => {
  let component: any;
  beforeEach(() => {
    component = setUp();
  });

  test('It should render without errors', (): void => {
    const wrapper = component.find('.ids-icon-btn');
    expect(wrapper.length).toBe(1);
  });
});
