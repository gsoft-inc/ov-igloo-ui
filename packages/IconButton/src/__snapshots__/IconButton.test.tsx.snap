// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton Component It should render a danger appearance 1`] = `
<DocumentFragment>
  <button
    class="ids-btn ids-icon-btn ids-icon-btn--medium ids-btn--danger"
    data-test="ids-icon-btn"
    title=""
    type="button"
  >
    <span
      class=""
    >
      <svg
        class="ids-icon"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 12V4M12 8H4"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;

exports[`IconButton Component It should render a default theme 1`] = `
<DocumentFragment>
  <button
    class="ids-btn ids-icon-btn ids-icon-btn--medium"
    data-test="ids-icon-btn"
    theme="default"
    title=""
    type="button"
  >
    <span
      class=""
    >
      <svg
        class="ids-icon"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 12V4M12 8H4"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;

exports[`IconButton Component It should render a ghost appearance 1`] = `
<DocumentFragment>
  <button
    class="ids-btn ids-icon-btn ids-icon-btn--medium ids-btn--ghost"
    data-test="ids-icon-btn"
    title=""
    type="button"
  >
    <span
      class=""
    >
      <svg
        class="ids-icon"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 12V4M12 8H4"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;

exports[`IconButton Component It should render a premium appearance 1`] = `
<DocumentFragment>
  <button
    class="ids-btn ids-icon-btn ids-icon-btn--medium ids-btn--premium"
    data-test="ids-icon-btn"
    title=""
    type="button"
  >
    <span
      class=""
    >
      <svg
        class="ids-icon"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 12V4M12 8H4"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;

exports[`IconButton Component It should render a primary appearance 1`] = `
<DocumentFragment>
  <button
    class="ids-btn ids-icon-btn ids-icon-btn--medium"
    data-test="ids-icon-btn"
    title=""
    type="button"
  >
    <span
      class=""
    >
      <svg
        class="ids-icon"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 12V4M12 8H4"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;

exports[`IconButton Component It should render a round theme 1`] = `
<DocumentFragment>
  <button
    class="ids-btn ids-icon-btn ids-icon-btn--medium"
    data-test="ids-icon-btn"
    theme="round"
    title=""
    type="button"
  >
    <span
      class=""
    >
      <svg
        class="ids-icon"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 12V4M12 8H4"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;

exports[`IconButton Component It should render a secondary appearance 1`] = `
<DocumentFragment>
  <button
    class="ids-btn ids-icon-btn ids-icon-btn--medium ids-btn--secondary"
    data-test="ids-icon-btn"
    title=""
    type="button"
  >
    <span
      class=""
    >
      <svg
        class="ids-icon"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 12V4M12 8H4"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;
